tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -155,32 +178,10 @@ module.exports = {
     changeCustomPermissionsTo: async function (collaborator, requiredPermissions) {
       const requiredPermissionArray = this.getArrayFromPermissionString(requiredPermissions)
 
-      let permission = ''
-      this.expandInformationSelector(collaborator)
-
-      for (let i = 0; i < collaboratorPermissionArray.length; i++) {
-        permission = collaboratorPermissionArray[i]
-        const permissionXpath = this.getPermissionSwitchXpath(permission)
-        // Check if the xpath of permission is visible
-        await this.api.waitForElementVisible({ selector: permissionXpath, timeout: 100, abortOnFailure: false }, function (result) {
-          if (result.value === true) {
-            this
-              .getAttribute(permissionXpath, 'data-state', (state) => {
-                if ((state.value === 'on' && !requiredPermissionArray.includes(permission)) ||
-                  (state.value === 'off' && requiredPermissionArray.includes(permission))) {
-                  // need to click
-                  this.useXpath()
-                    .click(permissionXpath)
-                }
-              })
-          }
-          // check if the requiredPermission is not visible
-          if (result.value === false && requiredPermissionArray.includes(permission)) {
-            throw new Error(`permission ${permission} is not visible `)
-          }
-        })
-      }
-      return this.saveCollaboratorPermission()
+      await this.expandInformationSelector(collaborator)
+      const promises = requiredPermissionArray.map(permission => this.changeSinglePermission(permission, true))
+      return Promise.all(promises)
```

Comment:
The refactor is not equivalent. Please do not run this in parallel. I prefer very much in running this in a synchronous manner. Please use `for ... await` as done before.